File: 1603-design-parking-system/1603-design-parking-system.py
class ParkingSystem:
    slot = [0, 0, 0]

    def __init__(self, big: int, medium: int, small: int):
        self.slot = [big, medium, small]

    def addCar(self, carType: int) -> bool:
        if self.slot[carType - 1] == 0:
            return False
        else:
            self.slot[carType -1 ]-=1
            return True


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)

File: 1603-design-parking-system/1603-design-parking-system.cpp
#include <vector>
class ParkingSystem {
public:
  vector<int> slot = {0, 0, 0};
  ParkingSystem(int big, int medium, int small) {
    this->slot = {big, medium, small};
  }

  bool addCar(int carType) {
    if (this->slot[carType - 1] == 0)
      return false;
    else {
      this->slot[carType - 1] -= 1;
      return true;
    }
  }
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */